tests/rest/client/test_sync.py

```diff
@@ -639,13 +653,74 @@ def test_unread_counts(self) -> None:
         self._check_unread_count(4)
 
         # Check that tombstone events changes increase the unread counter.
-        self.helper.send_state(
+        res1 = self.helper.send_state(
             self.room_id,
             EventTypes.Tombstone,
             {"replacement_room": "!someroom:test"},
             tok=self.tok2,
         )
         self._check_unread_count(5)
+        res2 = self.helper.send(self.room_id, "hello", tok=self.tok2)
+
+        # Make sure both m.read and org.matrix.msc2285.read.private advance
+        channel = self.make_request(
+            "POST",
+            f"/rooms/{self.room_id}/receipt/m.read/{res1['event_id']}",
+            {},
+            access_token=self.tok,
+        )
+        self.assertEqual(channel.code, 200, channel.json_body)
+        self._check_unread_count(1)
+
+        channel = self.make_request(
+            "POST",
+            f"/rooms/{self.room_id}/receipt/org.matrix.msc2285.read.private/{res2['event_id']}",
+            {},
+            access_token=self.tok,
+        )
+        self.assertEqual(channel.code, 200, channel.json_body)
+        self._check_unread_count(0)
+
+    @override_config({"experimental_features": {"msc2285_enabled": True}})
+    def test_read_receipts_only_go_down(self) -> None:
+        # Join the new user
+        self.helper.join(room=self.room_id, user=self.user2, tok=self.tok2)
+
+        # We tests this for both receipt types that influence notification counts
+        for receipt_type in [ReceiptTypes.READ, ReceiptTypes.READ_PRIVATE]:
```

Comment:
A better option for this might be to use `parameterized`, looks like this is used on the `test_read_receipt_with_empty_body` test.

Reply:
Ah, thanks! I was thinking there was a better way to do this! :sweat_smile: